### ISSUE TITLE
AssertDatabaseMatchesConfiguration throws exception when there's schematically no difference

### ISSUE THREAD

**Issue by isengrim613** [JasperFx/marten#1150]
I am writing my own schema creation/migration scripts to make use of postgres table partitioning to handle the large `mt_events` table when I encountered this.

It turns out that:
```
            string expected =
                @"CREATE OR REPLACE FUNCTION public.mt_upsert_account(doc JSONB, docDotNetType varchar, docId varchar, docVersion uuid) RETURNS UUID LANGUAGE plpgsql SECURITY INVOKER AS $function$
DECLARE
  final_version uuid;
BEGIN
  INSERT INTO public.mt_doc_account (""data"", ""mt_dotnet_type"", ""id"", ""mt_version"", mt_last_modified) VALUES (doc, docDotNetType, docId, docVersion, transaction_timestamp())
        ON CONFLICT ON CONSTRAINT pk_mt_doc_account
            DO UPDATE SET ""data"" = doc, ""mt_dotnet_type"" = docDotNetType, ""mt_version"" = docVersion, mt_last_modified = transaction_timestamp();

        SELECT mt_version FROM public.mt_doc_account into final_version WHERE id = docId ;
    RETURN final_version;
    END;
    $function$;";
```

is different from 

```
            string actual =
                @"CREATE OR REPLACE FUNCTION public.mt_upsert_account(doc JSONB, docDotNetType VARCHAR, docId VARCHAR, docVersion uuid)
  RETURNS UUID
  LANGUAGE plpgsql SECURITY INVOKER AS
$function$
DECLARE
  final_version uuid;
BEGIN
  INSERT INTO public.mt_doc_account (""data"", ""mt_dotnet_type"", ""id"", ""mt_version"", mt_last_modified)
        VALUES (doc, docDotNetType, docId, docVersion, transaction_timestamp())
        ON CONFLICT ON CONSTRAINT pk_mt_doc_account
            DO UPDATE SET ""data"" = doc, ""mt_dotnet_type"" = docDotNetType, ""mt_version"" = docVersion, mt_last_modified = transaction_timestamp();

        SELECT mt_version
        FROM public.mt_doc_account
        INTO final_version
        WHERE id = docId;
    RETURN final_version;
    END;
    $function$;";
```

Expected string after CanonicizeSql:
```
CREATE OR REPLACE FUNCTION public.mt_upsert_account(doc JSONB, docDotNetType varchar, docId varchar, docVersion uuid) RETURNS UUID DECLARE final_version uuid; BEGIN INSERT INTO public.mt_doc_account ("data", "mt_dotnet_type", "id", "mt_version", mt_last_modified) VALUES (doc, docDotNetType, docId, docVersion, transaction_timestamp()) ON CONFLICT ON CONSTRAINT pk_mt_doc_account DO UPDATE SET "data" = doc, "mt_dotnet_type" = docDotNetType, "mt_version" = docVersion, mt_last_modified = transaction_timestamp(); SELECT mt_version FROM public.mt_doc_account into final_version WHERE id = docId ; RETURN final_version; END; $function$
```

Actual string after CanonicizeSql
```
CREATE OR REPLACE FUNCTION public.mt_upsert_account(doc JSONB, docDotNetType VARCHAR, docId VARCHAR, docVersion uuid) RETURNS UUID DECLARE final_version uuid; BEGIN INSERT INTO public.mt_doc_account ("data", "mt_dotnet_type", "id", "mt_version", mt_last_modified) VALUES (doc, docDotNetType, docId, docVersion, transaction_timestamp()) ON CONFLICT ON CONSTRAINT pk_mt_doc_account DO UPDATE SET "data" = doc, "mt_dotnet_type" = docDotNetType, "mt_version" = docVersion, mt_last_modified = transaction_timestamp(); SELECT mt_version FROM public.mt_doc_account INTO final_version WHERE id = docId; RETURN final_version; END; $function$
```

The difference is a space before the semi-colon:
```
SELECT mt_version FROM public.mt_doc_account INTO final_version WHERE id = docId ;
```
vs 
```
SELECT mt_version FROM public.mt_doc_account INTO final_version WHERE id = docId;
```

Since spaces before semi-colons don't do anything, they should be removed in the `CanonicizeSql` function.

**Comment by isengrim613** [comment]
PR that will achieve this:
https://github.com/testcontainers/testcontainers-dotnet/issues/33

**Comment by isengrim613** [comment]
PR merged

**Comment by mysticmind** [comment]
Fixed via PR #1149
